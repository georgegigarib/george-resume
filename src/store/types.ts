export interface RootState {
    hash: {
      activeHash: string;
    };
  }
  
  export interface HashAction {
    type: string;
    payload: string;
  }
  