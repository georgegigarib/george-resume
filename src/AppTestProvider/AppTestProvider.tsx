import React from 'react'
import { I18nextProvider } from 'react-i18next'
import i18n from '@/locales/i18n.config'
import store from '@/store'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'

const AppTestProvider = ({ children }: { children: React.ReactNode }) => {
  return (
    <I18nextProvider i18n={i18n}>
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/" element={children} />
          </Routes>
        </Router>
      </Provider>
    </I18nextProvider>
  )
}

export default AppTestProvider
