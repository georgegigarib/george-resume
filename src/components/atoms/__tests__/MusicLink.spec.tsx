import { render, fireEvent } from '@testing-library/react'
import { describe, expect, it, vi } from 'vitest'
import MusicLink from '@/components/atoms/MusicLink'
import { Sections } from '@/app/layout/types/NavBarTypes'
import AppTestProvider from '@/AppTestProvider/AppTestProvider'

describe('MusicLink component', () => {
  const mockScrollToSection = vi.fn()
  const mockSetAutomaticScroll = vi.fn()

  const renderComponent = (activeHash: string) =>
    render(
      <AppTestProvider>
        <MusicLink
          scrollToSection={mockScrollToSection}
          setAutomaticScroll={mockSetAutomaticScroll}
          activeHash={activeHash}
        />
      </AppTestProvider>
    )

  it('matches the snapshot when activeHash is Sections.Dev', () => {
    const { asFragment } = renderComponent(Sections.Dev)
    expect(asFragment()).toMatchSnapshot()
  })

  it('matches the snapshot when activeHash is Sections.Music', () => {
    const { asFragment } = renderComponent(Sections.Music)
    expect(asFragment()).toMatchSnapshot()
  })

  it('matches the snapshot when activeHash is Sections.Me', () => {
    const { asFragment } = renderComponent(Sections.Me)
    expect(asFragment()).toMatchSnapshot()
  })

  it('calls scrollToSection and setAutomaticScroll on click', () => {
    const { getByText } = renderComponent(Sections.Music)
    const linkElement = getByText('music')

    fireEvent.click(linkElement)

    expect(mockScrollToSection).toHaveBeenCalledWith(Sections.Music, true)
    expect(mockSetAutomaticScroll).toHaveBeenCalledWith(false)
  })
})
