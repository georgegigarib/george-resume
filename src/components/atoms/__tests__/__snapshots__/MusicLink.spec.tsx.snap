// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MusicLink component > matches the snapshot when activeHash is Sections.Dev 1`] = `
<DocumentFragment>
  <a
    class="block rounded md:p-0 -translate-y-[60%] md:-translate-y-[80%] 2xl:-translate-y-[50%] animated-transform"
  >
    music
  </a>
</DocumentFragment>
`;

exports[`MusicLink component > matches the snapshot when activeHash is Sections.Me 1`] = `
<DocumentFragment>
  <a
    class="block rounded md:p-0 text-gray-900 -rotate-[17deg] -translate-y-[20%] sm:-rotate-12 md:translate-y-[0%] xl:-rotate-3 animated-transform opacity-40 font-bold 2xl:translate-y-[20%]"
  >
    music
  </a>
</DocumentFragment>
`;

exports[`MusicLink component > matches the snapshot when activeHash is Sections.Music 1`] = `
<DocumentFragment>
  <a
    class="block rounded md:p-0 text-white font-bold animated-transform translate-y-[15%] md:translate-y-[45%] scale-130 -rotate-5"
  >
    music
  </a>
</DocumentFragment>
`;
